patching/plugin-commands-patching/src/getPatchedDependency.ts

```diff
@@ -0,0 +1,63 @@
+import path from 'path'
+import { parseWantedDependency, ParseWantedDependencyResult } from '@pnpm/parse-wanted-dependency'
+import { prompt } from 'enquirer'
+import { readCurrentLockfile } from '@pnpm/lockfile-file'
+import { nameVerFromPkgSnapshot } from '@pnpm/lockfile-utils'
+import { PnpmError } from '@pnpm/error'
+import { WANTED_LOCKFILE } from '@pnpm/constants'
+import semver from 'semver'
+
+export async function getPatchedDependency (rawDependency: string, lockfileDir: string): Promise<ParseWantedDependencyResult> {
+  const dep = parseWantedDependency(rawDependency)
+
+  const { versions, preferredVersions } = await getVersionsFromLockfile(dep, lockfileDir)
+
+  if (!preferredVersions.length) {
+    throw new PnpmError(
+      'PATCH_VERSION_NOT_FOUND',
+      `Can not find ${rawDependency} in project ${lockfileDir}, ${versions.length ? `you can specify currently installed version: ${versions.join(', ')}.` : `did you forget to install ${rawDependency}?`}`
+    )
+  }
+
+  dep.alias = dep.alias ?? rawDependency
+  if (preferredVersions.length > 1) {
+    const { version } = await prompt<{
+      version: string
+    }>({
+      type: 'select',
+      name: 'version',
+      message: 'Choose which version to patch',
+      choices: versions,
+    })
+    dep.pref = version
+  } else {
+    dep.pref = preferredVersions[0]
+  }
+
+  return dep
+}
+
+async function getVersionsFromLockfile (dep: ParseWantedDependencyResult, lockfileDir: string) {
+  const lockfile = (await readCurrentLockfile(path.join(lockfileDir, 'node_modules/.pnpm'), {
```

Comment:
this is not so simple. There are 2 more options that control where the internal lockfile is located: `modulesDir` and `virtualStoreDir`.

https://github.com/pnpm/pnpm/blob/787c43dcc1e913062a916c5e91440865bd9a8a6d/pkg-manager/get-context/src/index.ts#L102-L104